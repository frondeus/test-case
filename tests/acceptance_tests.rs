#![cfg(test)]

mod acceptance {
    use itertools::Itertools;
    use std::env;
    use std::path::PathBuf;
    use std::process::Command;

    #[test]
    fn basic() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("basic"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn hamcrest_assertions() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("hamcrest_assertions"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn r#async() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("async"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }
}
