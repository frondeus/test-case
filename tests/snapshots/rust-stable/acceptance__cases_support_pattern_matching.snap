---
source: tests/acceptance_tests.rs
assertion_line: 131
expression: output
---
    extended_pattern_matching_result::err_should_fail
    extended_pattern_matching_result::ok_should_fail
---- extended_pattern_matching_result::err_should_fail stdout ----
---- extended_pattern_matching_result::ok_should_fail stdout ----
error: test failed, to rerun pass '--lib'
failures:
failures:
running 5 tests
test extended_pattern_matching_result::err_should_fail ... FAILED
test extended_pattern_matching_result::ok_should_fail ... FAILED
test extended_pattern_matching_result::simpleenum_var1_expects_matching_ok_e_e_simpleenum_var1 ... ok
test extended_pattern_matching_result::simpleenum_var2_expects_matching_err_e_e_var2_ ... ok
test pattern_matching_result::simpleenum_var2_expects_matching_simpleenum_var2_ ... ok
test result: FAILED. 3 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
thread 'extended_pattern_matching_result::err_should_fail' panicked at 'Expected `Err(e) if e == "var1"` found Err("var2")', src/lib.rs:16:1
thread 'extended_pattern_matching_result::ok_should_fail' panicked at 'Expected `Ok(e) if e == SimpleEnum :: Var2` found Ok(Var1)', src/lib.rs:16:1
